libs/execution_context/ebpf_link.c

```diff
@@ -469,3 +484,36 @@ ebpf_link_get_info(
     *info_size = sizeof(*info);
     EBPF_RETURN_RESULT(EBPF_SUCCESS);
 }
+
+/**
+ * @brief Set and validate the link state.
+ *
+ * @param[in] link Link to set state on.
+ * @param[in] new_state New state to set.
+ */
+_Requires_lock_held_(link->attach_lock) static void _ebpf_link_set_state(
+    _Inout_ ebpf_link_t* link, ebpf_link_state_t new_state)
+{
+    ebpf_link_state_t old_state = link->state;
```

Comment:
https://github.com/microsoft/ebpf-for-windows/actions/runs/4821567423/jobs/8587615251?pr=2396 fails because old_state is not used in a release build.  Suggest deleting this variable and just using `link->state` where old_state is use below.